charts/testkube-cloud-api/templates/deployment.yaml

```diff
@@ -326,10 +326,16 @@ spec:
               value: "{{ .Values.api.features.bootstrapConfig.enabled}}"
             - name: "BOOTSTRAP_CONFIG_PATH"
               value: "/app/config/config"
+            - name: "TESTKUBE_DEFAULT_AGENT_TOKEN"
```

Comment:
I cannot find this environment variable in the source for the cloud API. What does this do? Is this something you are implementing on a fork?

Reply:
It's a bit complicated

This is an env related to the bootstrap ConfigMap. Basically, we want to lift up the Agent Token to a shared Kubernetes Secret to boot both components without dependency. That means that both cloud-api and agent need to be able to read the token from the secret.

In an ideal Kubernetes-native world you would pass the `secretKeyRef` to the component and that process natively interacts with Kubernetes to read the secret. We wanted to avoid having to introduce the role to read Secret in cloud-api. Instead we work with an indirection, the so called `envRef`. We then instead mount the secret to the env and we can simply read from env. It's a dynamic ref because you can have multiple environments.

```yaml
# boot.yaml
organizations:
  - environments:
      - agentToken:
          valueFrom:
            envRef: TESTKUBE_DEFAULT_AGENT_TOKEN
        name: default
    name: far-strider
```